src/stream/src/executor/mview/materialize.rs

```diff
@@ -58,6 +58,21 @@ pub struct MaterializeExecutor<S: StateStore, SD: ValueRowSerde> {
     conflict_behavior: ConflictBehavior,
 
     version_column_index: Option<u32>,
+
+    may_have_downstream: bool,
```

Comment:
Why this is `may_have_downstream`, instead of `has_downstream`?

Reply:
The flag only gets changed when there is newly created downstream, and won't be changed when some downstreams are dropped, so the value can be false positive.

Reply:
Sure. That's the same with my understanding. But are there any problems if we also handle the drop and re-enable the switch?

@BugenZhao mentioned this optimization
https://github.com/risingwavelabs/risingwave/pull/16348#discussion_r1577302916

Reply:
Previously we cannot do this because there is no enough information in the barrier with drop streaming job command. @chenzl25 Can you provide with more details?

But I think it can be implemented if we incrementally maintain a full copy of downstream information in the mv executor.

Reply:
Take `AddMutation` as an example. It contains `pub adds: HashMap<ActorId, Vec<PbDispatcher>>` to represent how many dispatcher to be added for a actor, but for `UpdateMutation`, it only has `dropped_actors: HashSet<ActorId>` so we don't have enough information to maintain the dispatcher number for `MaterializedView`.